src/app/services/params/params.service.ts

```diff
@@ -0,0 +1,97 @@
+import { inject, Injectable } from '@angular/core';
+import { DateFormat, DateRange, QueryParams } from '../../models';
+import { DateService } from '../date';
+
+/**
+ * Gets and sets params without changing navigation history.
+ * Because this, it doesn't work with the `Router` Service.
+ */
+@Injectable({
+	providedIn: 'root',
+})
+export class ParamsService {
+	private readonly dateService = inject(DateService);
+
+	/**
+	 * Gets param value by name
+	 */
+	getParam(name: string): string | null {
+		try {
+			const params = new URLSearchParams(window.location.search);
+
+			return params.get(name);
+		} catch (error) {
+			console.error(error);
+
+			return null;
+		}
+	}
+
+	/**
+	 * Set param value by name. Replaces any previous value(s) currently in the url
+	 */
+	setParam(name: string, value: string): void {
+		// Get current params
+		const params = new URLSearchParams(window.location.search);
+
+		// Clear any value if there is already one
+		// If you append a value without deleting, there will be duplicate name value pairs in the url
+		params.delete(name);
+
+		params.append(name, value);
+		params.sort();
+
+		const baseUrl = window.location.href.split('?')[0];
+
+		// Update params without changing navigation history
+		window.history.replaceState({}, document.title, `${baseUrl}?${params.toString()}`);
+	}
+
+	/**
+	 * Get package names in params
+	 */
+	getPackageNames(): string[] {
+		const packageNames = this.getParam(QueryParams.PACKAGE_NAMES)?.split(',') ?? [];
+
+		// Filter any empty strings since those wouldn't be valid package names
+		return packageNames.filter((packageName) => !!packageName);
+	}
+
+	/**
+	 * Set package names in params
+	 */
+	setPackageNames(dateRange: string[]): void {
+		this.setParam(QueryParams.PACKAGE_NAMES, dateRange.toString());
+	}
+
+	/**
+	 * Get date range in params
+	 */
+	getDateRange(): DateRange | null {
+		const dateRangeParams = this.getParam(QueryParams.DATE_RANGE)?.split(',') ?? [];
+
+		// Ensure that there's exactly 2 values for the date range
+		if (dateRangeParams.length !== 2) {
+			return null;
+		}
+
+		const [start, end] = dateRangeParams;
+
+		return [
+			this.dateService.getDate(start, DateFormat.YEAR_MONTH_DAY),
+			this.dateService.getDate(end, DateFormat.YEAR_MONTH_DAY),
+		];
```

Comment:
Don't we want to implement a check here to verify that the `start` date in router params is before the `end` date?
To prevent manually modifying the URL and sending it to somebody

Reply:
Hmm, I'm unsure if we want to do the logical validation here for something like that. But it seems fair.

I know that it's not checked directly for `DateService.getDateRange`, but maybe that validation logic can be abstracted to `DateService` and we can use that in both situations. I might rename `DateService.getDateRange` to `DateService.getDates` to avoid confusion on what a "DateRange" is (is it a full list of Dates between start and end or is it just the start and end)

Reply:
I believe renaming it to `getDates()` would make more sense. I guess we can also implement the validation (start less than end) in `DateService.validateStartBeforeEnd(start, end)` and in the code they would be called like:
`DateService. getDates()`
`DateService. validateStartBeforeEnd()`
something like that